experimental/PropertyDDS/packages/property-dds/src/propertyTreeExtFactories.ts

```diff
@@ -2,68 +2,133 @@
  * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
  * Licensed under the MIT License.
  */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
 import { deflate, inflate } from "pako";
+import { compress, decompress } from "lz4js";
 import { bufferToString, stringToBuffer } from "@fluidframework/common-utils";
 import {
     IChannelAttributes,
     IFluidDataStoreRuntime,
     IChannelServices,
     IChannelFactory,
+    IChannel,
 } from "@fluidframework/datastore-definitions";
-import { IPropertyTreeMessage, ISharedPropertyTreeEncDec, ISnapshotSummary, SharedPropertyTreeOptions }
-from "./propertyTree";
-import { DeflatedPropertyTree } from "./propertyTreeExt";
-
-function encodeSummary(snapshotSummary: ISnapshotSummary) {
-    const summaryStr = JSON.stringify(snapshotSummary);
-    const unzipped = new TextEncoder().encode(summaryStr);
-    const serializedSummary: Buffer = deflate(unzipped);
-    return serializedSummary;
+import {
+    IPropertyTreeConfig, IPropertyTreeMessage, ISharedPropertyTreeEncDec,
+    ISnapshotSummary, SharedPropertyTree, SharedPropertyTreeOptions,
 }
+    from "./propertyTree";
+import { DeflatedPropertyTree, LZ4PropertyTree } from "./propertyTreeExt";
 
-function decodeSummary(serializedSummary): ISnapshotSummary {
-    const unzipped = inflate(serializedSummary);
-    const summaryStr = new TextDecoder().decode(unzipped);
-    const snapshotSummary: ISnapshotSummary = JSON.parse(summaryStr);
-    return snapshotSummary;
-}
+/**
+ * This class contains builders of the compression methods used to compress
+ * of summaries and messages with the plugable compression algorithm.
+ */
+class CompressionMethods {
```

Comment:
Most JS developers would find a function more natural than a class here:

function createCompressionMethods(encoder, decoder) {
        return {
            messageEncoder: {
                encode: (...) => { ... },
                decode: (...) => { ... },
            },
            summaryEncoder: {
                encode: (...) => { ... },
                decode: (...) => { ... },
            },
        };
    };
    
Possibly this function should be part of CompressedPropertyTreeFactory.

Reply:
@DLehenbauer The idea here is complete separation of compression methods from the CompressedPropertyTreeFactory. The only relevant method here is getEncDec and how it is implemented is the responsibility of the extended Factory class. CompressionMethods is only a helper, just a utility class which helps the extended factories to reduce the code if they want this behavior but that is only one way of many. Puristic approach would require here the method getEncDec to be abstract. I have chosen the default behavior using the CompressionMethods but did not want to bind them too closely with CompressedPropertyTreeFactory so I kept them outside in another util class. 

Concerning the method usage instead of class usage, I can do it but it looks to me that it might produce messy code putting various responsibilities in one place (one method). I was trying to follow single responsibility pattern and separate the functionality in dedicated methods referencing them in the returned object. I could create 5 functions instead of the class for building encoders, decoders and createCompressionMethods but it looked to me more self explaining if I put them to one class.

Nevertheless I will follow your suggestions if you still think that it is more reasonable to do it by function within the CompressedPropertyTreeFactory.